apps/jobs/views.py

```diff
@@ -538,66 +542,117 @@ def get_submission_by_pk(request, submission_id):
+                challenge_phase_split = ChallengePhaseSplit.objects.get(
+                    challenge_phase__pk=challenge_phase_pk,
+                    dataset_split__codename=split)
+            except ChallengePhaseSplit.DoesNotExist:
+                response_data = {'error': 'Challenge Phase Split does not exist with phase_id: {} and split codename: {}'.format(challenge_phase_pk, split)}
+                return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+            leaderboard_metrics = challenge_phase_split.leaderboard.schema.get('labels')
+            missing_metrics = []
+            malformed_metrics = []
+            for metric, value in accuracies.items():
+                if metric not in leaderboard_metrics:
+                    missing_metrics.append(metric)
+
+                if not (isinstance(value, float) or isinstance(value, int)):
+                    malformed_metrics.append((metric, type(value)))
+
+            if len(missing_metrics):
+                response_data = {'error': 'Following metrics are missing in the leaderboard data: {}'.format(missing_metrics)}
+                return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+            if len(malformed_metrics):
+                response_data = {'error': 'Values for following metrics are not of float/int: {}'.format(malformed_metrics)}
+                return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+            serializer = CreateLeaderboardDataSerializer(
+                data=request.data.copy(),
+                context={
+                    'challenge_phase_split': challenge_phase_split,
+                    'submission': submission,
+                    'request': request,
+                }
+            )
+            if serializer.is_valid():
+                leaderboard_data_list.append(serializer)
+            else:
+                return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
-        if serializer.is_valid():
-            serializer.save()
-        else:
-            return Response(serializer.errors, status=status.HTTP_406_NOT_ACCEPTABLE)
+        try:
+            with transaction.atomic():
+                for serializer in leaderboard_data_list:
+                    serializer.save()
+        except IntegrityError:
+            logger.exception('Failed to update submission_id {} related metadata'.format(submission_pk))
+            response_data = {'error': 'Failed to update submission_id {} related metadata'.format(submission_pk)}
+            return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
 
     submission.status = submission_status
     submission.completed_at = timezone.now()
     submission.stdout_file.save('stdout.txt', ContentFile(stdout_content))
     submission.stderr_file.save('stderr.txt', ContentFile(stderr_content))
     submission.submission_result_file.save('submission_result.json', ContentFile(submission_result))
```

Comment:
@deshraj Just wanted to confirm, This will output all the data in the results into the `results.json` file which will be visible to participants.
As of now, we don't show participants their accuracies on a particular split but this will generate a file with content like
```
[{"split":"split2","accuracies":{"score":50}},{"split":"split1","accuracies":{"score":50}}]
```
into it. I think this isn't the expected bahaviour.

Reply:
Good catch Rishabh. I am glad that we caught it before it could have been deployed to any challenge. I will add a key `show_to_participant` in the API and that will take care of this.